### PR TITLE
reflection probe support instancing

### PR DESCRIPTION
**Pull request by xubing0906**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by xubing0906** [comment]
@cocos-robot run test cases

**Comment by xubing0906** [comment]
@cocos-robot run test cases

**Comment by xubing0906** [comment]
@cocos-robot run test cases

**Comment by xubing0906** [comment]
@cocos-robot run test cases

File: cocos/3d/framework/mesh-renderer.ts [hunk 1]
```diff
@@ -653,22 +653,27 @@ export class MeshRenderer extends ModelRenderer {
...
         }
     }
 
-    protected _onUpdateReflectionProbeTexture () {
-        if (this.model === null) return;
+    protected _updateReflectionProbeTexture () {
+        if (this._model === null) return;
         if (this.bakeSettings.reflectionProbe === ReflectionProbeType.BAKED_CUBEMAP) {
-            this.model.updateReflctionProbeCubemap(this.bakeSettings._probeCubemap!);
+            this._model.updateReflctionProbeCubemap(this.bakeSettings._probeCubemap);
```
Comment (jk20012001) [inline cocos/3d/framework/mesh-renderer.ts]: Reflection

File: cocos/rendering/reflection-probe/reflection-probe-stage.ts [hunk 2]
```diff
@@ -107,9 +110,17 @@ export class ReflectionProbeStage extends RenderStage {
...
 
         if (this._probe!.camera.clearFlag & ClearFlagBit.COLOR) {
-            colors[0].x = this._probe!.camera.clearColor.x;
-            colors[0].y = this._probe!.camera.clearColor.y;
-            colors[0].z = this._probe!.camera.clearColor.z;
+            this._rgbeColor.x = this._probe!.camera.clearColor.x;
+            this._rgbeColor.y = this._probe!.camera.clearColor.y;
+            this._rgbeColor.z = this._probe!.camera.clearColor.z;
+            let rgbe = this._probe!.camera.clearColor;
+            if (this._probe!.probeType === ProbeType.CUBE) {
```
Comment (jk20012001) [inline cocos/rendering/reflection-probe/reflection-probe-stage.ts]: planar reflection also needs rgbe

**Review by jk20012001 — APPROVED** [review]

**Pull request merged**